Could not find :darcula:1.0.0

My build failes with this message:

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':shadowJar'.
> Could not resolve all dependencies for configuration ':runtimeClasspath'.
   > Could not find :darcula:1.0.0.
     Searched in the following locations:
       - https://raw.githubusercontent.com/bulenkov/Darcula/master/build/darcula.xml
     Required by:
         project :
```

Info:
MacOs 10.15.2

java 13.0.1 2019-10-15
Java(TM) SE Runtime Environment (build 13.0.1+9)
Java HotSpot(TM) 64-Bit Server VM (build 13.0.1+9, mixed mode, sharing)

Gradle 6.3
------------------------------------------------------------

Build time:   2020-03-24 19:52:07 UTC
Revision:     bacd40b727b0130eeac8855ae3f9fd9a0b207c60

Kotlin:       1.3.70
Groovy:       2.5.10
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          13.0.2 (Oracle Corporation 13.0.2+8)
OS:           Mac OS X 10.15.2 x86_64

--- a/build.gradle
+++ b/build.gradle
@@ -76,10 +76,10 @@ configurations {
 dependencies {
     compile 'com.google.guava:guava:28.0-jre'
     compile 'com.google.code.gson:gson:2.8.5'
-    compile 'org.ow2.asm:asm:7.1'
-    compile 'org.ow2.asm:asm-commons:7.1'
-    compile 'org.ow2.asm:asm-tree:7.1'
-    compile 'org.ow2.asm:asm-util:7.1'
+    compile 'org.ow2.asm:asm:8.0'
+    compile 'org.ow2.asm:asm-commons:8.0'
+    compile 'org.ow2.asm:asm-tree:8.0'
+    compile 'org.ow2.asm:asm-util:8.0'
     compile 'net.sf.jopt-simple:jopt-simple:6.0-alpha-3'
 
     compile 'net.fabricmc:procyon-fabric-compilertools:0.5.35.+'

--- a/build.gradle
+++ b/build.gradle
@@ -19,7 +19,7 @@ apply plugin: 'com.github.johnrengelman.shadow'
 apply plugin: 'maven'
 
 group = 'cuchaz'
-version = '0.15'
+version = '0.15.1'
 
 def generatedSourcesDir = "$buildDir/generated-src"
 

--- a/build.gradle
+++ b/build.gradle
@@ -24,9 +24,7 @@ version = '0.15.1'
 def generatedSourcesDir = "$buildDir/generated-src"
 
 def ENV = System.getenv()
-if (ENV.BUILD_NUMBER) {
-    version = version + "." + "${System.getenv().BUILD_NUMBER}"
-}
+version = version + "+" + (ENV.BUILD_NUMBER ? ("build." + ENV.BUILD_NUMBER) : "local")
 
 sourceCompatibility = 1.8
 targetCompatibility = 1.8

--- a/build.gradle
+++ b/build.gradle
@@ -60,6 +60,9 @@ repositories {
 	ivy {
 		name "darcula"
 		artifactPattern "https://raw.githubusercontent.com/bulenkov/Darcula/master/build/[module].[ext]"
+		metadataSources {
+			artifact()
+		}
 	}
 }
 
@@ -88,8 +91,10 @@ dependencies {
 
     testCompile 'junit:junit:4.+'
     testCompile 'org.hamcrest:hamcrest-all:1.+'
-    proGuard 'net.sf.proguard:proguard-base:5.+'
+    proGuard 'net.sf.proguard:proguard-base:6.+'
 }
+
+def libraryJarsArg = JavaVersion.current().java9Compatible ? "<java.home>/jmods": "<java.home>/lib/rt.jar"
 
 // For each set of test inputs, create an output jar and obfuscate it.
 file('src/test/java/cuchaz/enigma/inputs').listFiles().each { theFile ->
@@ -110,8 +115,8 @@ file('src/test/java/cuchaz/enigma/inputs').listFiles().each { theFile ->
             classpath configurations.proGuard
 
             args '@src/test/resources/proguard-test.conf', '-injars', file('build/test-inputs/' +
-                    "${theFile.name}.jar"), '-outjars', file('build/test-obf/' +
-                    "${theFile.name}.jar")
+                    "${theFile.name}.jar"), '-libraryjars', libraryJarsArg,
+                    '-outjars', file('build/test-obf/' + "${theFile.name}.jar")
         }
 
         test.dependsOn "${theFile.name}TestObf"

--- a/src/main/java/cuchaz/enigma/EnigmaProject.java
+++ b/src/main/java/cuchaz/enigma/EnigmaProject.java
@@ -18,6 +18,7 @@ import cuchaz.enigma.translation.representation.entry.LocalVariableEntry;
 import cuchaz.enigma.translation.representation.entry.MethodEntry;
 import cuchaz.enigma.utils.I18n;
 
+import cuchaz.enigma.utils.Utils;
 import org.objectweb.asm.ClassWriter;
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.tree.ClassNode;
@@ -152,7 +153,7 @@ public class EnigmaProject {
 					ClassNode node = classCache.getClassNode(entry.getFullName());
 					if (node != null) {
 						ClassNode translatedNode = new ClassNode();
-						node.accept(new TranslationClassVisitor(deobfuscator, Opcodes.ASM5, new SourceFixVisitor(Opcodes.ASM5, translatedNode, jarIndex)));
+						node.accept(new TranslationClassVisitor(deobfuscator, Utils.ASM_VERSION, new SourceFixVisitor(Utils.ASM_VERSION, translatedNode, jarIndex)));
 						return translatedNode;
 					}
 

--- a/src/main/java/cuchaz/enigma/analysis/BuiltinPlugin.java
+++ b/src/main/java/cuchaz/enigma/analysis/BuiltinPlugin.java
@@ -12,6 +12,7 @@ import cuchaz.enigma.translation.representation.entry.ClassEntry;
 import cuchaz.enigma.translation.representation.entry.Entry;
 import cuchaz.enigma.translation.representation.entry.FieldEntry;
 import cuchaz.enigma.utils.Pair;
+import cuchaz.enigma.utils.Utils;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.FieldVisitor;
@@ -69,7 +70,7 @@ public final class BuiltinPlugin implements EnigmaPlugin {
 		private final List<MethodNode> classInits = new ArrayList<>();
 
 		EnumFieldNameFindingVisitor(Map<Entry<?>, String> mappings) {
-			super(Opcodes.ASM7);
+			super(Utils.ASM_VERSION);
 			this.mappings = mappings;
 		}
 

--- a/src/main/java/cuchaz/enigma/analysis/ClassCache.java
+++ b/src/main/java/cuchaz/enigma/analysis/ClassCache.java
@@ -7,6 +7,7 @@ import cuchaz.enigma.ClassProvider;
 import cuchaz.enigma.ProgressListener;
 import cuchaz.enigma.analysis.index.JarIndex;
 import cuchaz.enigma.bytecode.translators.LocalVariableFixVisitor;
+import cuchaz.enigma.utils.Utils;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.Opcodes;
@@ -77,7 +78,7 @@ public final class ClassCache implements AutoCloseable, ClassProvider {
 
 		ClassNode node = new ClassNode();
 
-		LocalVariableFixVisitor visitor = new LocalVariableFixVisitor(Opcodes.ASM5, node);
+		LocalVariableFixVisitor visitor = new LocalVariableFixVisitor(Utils.ASM_VERSION, node);
 		reader.accept(visitor, 0);
 
 		return node;

--- a/src/main/java/cuchaz/enigma/analysis/IndexSimpleVerifier.java
+++ b/src/main/java/cuchaz/enigma/analysis/IndexSimpleVerifier.java
@@ -5,6 +5,7 @@ import cuchaz.enigma.analysis.index.InheritanceIndex;
 import cuchaz.enigma.translation.representation.AccessFlags;
 import cuchaz.enigma.translation.representation.entry.ClassDefEntry;
 import cuchaz.enigma.translation.representation.entry.ClassEntry;
+import cuchaz.enigma.utils.Utils;
 import org.objectweb.asm.Type;
 import org.objectweb.asm.tree.analysis.BasicValue;
 import org.objectweb.asm.tree.analysis.SimpleVerifier;
@@ -17,7 +18,7 @@ public class IndexSimpleVerifier extends SimpleVerifier {
     private final InheritanceIndex inheritanceIndex;
 
     public IndexSimpleVerifier(EntryIndex entryIndex, InheritanceIndex inheritanceIndex) {
-        super(ASM7, null, null, null, false);
+        super(Utils.ASM_VERSION, null, null, null, false);
         this.entryIndex = entryIndex;
         this.inheritanceIndex = inheritanceIndex;
     }

--- a/src/main/java/cuchaz/enigma/analysis/InterpreterPair.java
+++ b/src/main/java/cuchaz/enigma/analysis/InterpreterPair.java
@@ -1,5 +1,6 @@
 package cuchaz.enigma.analysis;
 
+import cuchaz.enigma.utils.Utils;
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.Type;
 import org.objectweb.asm.tree.AbstractInsnNode;
@@ -16,7 +17,7 @@ public class InterpreterPair<V extends Value, W extends Value> extends Interpret
     private final Interpreter<W> right;
 
     public InterpreterPair(Interpreter<V> left, Interpreter<W> right) {
-        super(Opcodes.ASM7);
+        super(Utils.ASM_VERSION);
         this.left = left;
         this.right = right;
     }

--- a/src/main/java/cuchaz/enigma/analysis/index/JarIndex.java
+++ b/src/main/java/cuchaz/enigma/analysis/index/JarIndex.java
@@ -22,6 +22,7 @@ import cuchaz.enigma.translation.representation.Lambda;
 import cuchaz.enigma.translation.representation.entry.*;
 import cuchaz.enigma.utils.I18n;
 
+import cuchaz.enigma.utils.Utils;
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.Opcodes;
 
@@ -63,10 +64,10 @@ public class JarIndex implements JarIndexer {
 		progress.init(4, I18n.translate("progress.jar.indexing"));
 
 		progress.step(1, I18n.translate("progress.jar.indexing.entries"));
-		classCache.visit(() -> new IndexClassVisitor(this, Opcodes.ASM5), ClassReader.SKIP_CODE);
+		classCache.visit(() -> new IndexClassVisitor(this, Utils.ASM_VERSION), ClassReader.SKIP_CODE);
 
 		progress.step(2, I18n.translate("progress.jar.indexing.references"));
-		classCache.visit(() -> new IndexReferenceVisitor(this, entryIndex, inheritanceIndex, Opcodes.ASM5), 0);
+		classCache.visit(() -> new IndexReferenceVisitor(this, entryIndex, inheritanceIndex, Utils.ASM_VERSION), 0);
 
 		progress.step(3, I18n.translate("progress.jar.indexing.methods"));
 		bridgeMethodIndex.findBridgeMethods();

--- a/src/main/java/cuchaz/enigma/bytecode/translators/TranslationSignatureVisitor.java
+++ b/src/main/java/cuchaz/enigma/bytecode/translators/TranslationSignatureVisitor.java
@@ -1,5 +1,6 @@
 package cuchaz.enigma.bytecode.translators;
 
+import cuchaz.enigma.utils.Utils;
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.signature.SignatureVisitor;
 
@@ -13,7 +14,7 @@ public class TranslationSignatureVisitor extends SignatureVisitor {
 	private final Stack<String> classStack = new Stack<>();
 
 	public TranslationSignatureVisitor(Function<String, String> remapper, SignatureVisitor sv) {
-		super(Opcodes.ASM5);
+		super(Utils.ASM_VERSION);
 		this.remapper = remapper;
 		this.sv = sv;
 	}

--- a/src/main/java/cuchaz/enigma/gui/GuiController.java
+++ b/src/main/java/cuchaz/enigma/gui/GuiController.java
@@ -111,7 +111,7 @@ public class GuiController {
 			}
 
 			ClassNode fixedNode = new ClassNode();
-			node.accept(new SourceFixVisitor(Opcodes.ASM7, fixedNode, project.getJarIndex()));
+			node.accept(new SourceFixVisitor(Utils.ASM_VERSION, fixedNode, project.getJarIndex()));
 			return fixedNode;
 		}, new SourceSettings(true, true));
 	}

--- a/src/main/java/cuchaz/enigma/utils/Utils.java
+++ b/src/main/java/cuchaz/enigma/utils/Utils.java
@@ -12,6 +12,7 @@
 package cuchaz.enigma.utils;
 
 import com.google.common.io.CharStreams;
+import org.objectweb.asm.Opcodes;
 
 import javax.swing.*;
 import java.awt.*;
@@ -30,6 +31,8 @@ import java.util.StringJoiner;
 import java.util.stream.Collectors;
 
 public class Utils {
+	
+	public static final int ASM_VERSION = Opcodes.ASM8;
 
 	public static int combineHashesOrdered(Object... objs) {
 		final int prime = 67;

--- a/src/test/resources/proguard-build.conf
+++ b/src/test/resources/proguard-build.conf
@@ -1,4 +1,3 @@
--libraryjars  <java.home>/lib/rt.jar
 -dontoptimize
 -dontobfuscate
 -dontwarn

--- a/src/test/resources/proguard-test.conf
+++ b/src/test/resources/proguard-test.conf
@@ -1,4 +1,3 @@
--libraryjars  <java.home>/lib/rt.jar
 -overloadaggressively
 -repackageclasses
 -allowaccessmodification

